baca2PackageManager/judge.py

```diff
@@ -0,0 +1,139 @@
+
+from typing import Any
+from enum import Enum
+from abc import ABC, abstractmethod
+
+
+class JudgeVerdict(Enum):
+    OK = 0
+    FAIL = 1
+    INCONCLUSIVE = 2
+
+
+class JudgeNodeBase(ABC):
+
+    def __init__(self, name: str):
+        self._id = name
+
+    def __hash__(self):
+        return hash(self.name)
+
+    @property
+    def name(self) -> str:
+        """:returns: a string value used for hashing an instance"""
+        return self._id
+
+    @abstractmethod
+    def start(self, *args, **kwargs) -> Any:
+        ...
+
+    @abstractmethod
+    def receive(self, *args, **kwargs) -> JudgeVerdict:
+        ...
+
+
+class EndNode(JudgeNodeBase):
+
+    def __init__(self, name: str, end_node_meaning: JudgeVerdict):
+        super().__init__(name)
+        self.meaning = end_node_meaning
+
+    def start(self, *args, **kwargs) -> Any:
```

Comment:
Nie wiem, czy przypadkiem nie będzie lepiej zrobić, żeby po starcie po prostu zostawał pass. To jest bardziej sytuacja, że coś kieruje nas do Node'a końcowego i Manager nie zastanawia się już, czy może go uruchomić, czy nie, tylko po prostu puszcza start. To jest raczej w drugą stronę - dla tego Node'a nigdy nie wystąpi receive - jak coś wchodzi w stan końcowy to już z niego nie wychodzi (chociaż od tej reguły też mogą być potencjalne wyjątki)